exporters/otlp/src/otlp_http_client.cc

```diff
@@ -65,6 +50,146 @@ namespace otlp
+                                Base64EscapeChars &base64_enc_map,
+                                unsigned char padding_char)
+{
+  size_t i, n, nopadding;
+  int C1, C2, C3;
+  unsigned char *p;
+
+  if (slen == 0)
+  {
+    *olen = 0;
+    return (0);
+  }
+
+  n = (slen + 2) / 3;
+
+  if (n > (BASE64_SIZE_T_MAX - 1) / 4)
+  {
+    *olen = BASE64_SIZE_T_MAX;
+    return -1;
+  }
+
+  n *= 4;
+
+  // no padding
+  if (0 == padding_char)
+  {
+    nopadding = slen % 3;
+    if (0 != nopadding)
+    {
+      n -= 3 - nopadding;
+    }
+  }
+
+  if ((dlen < n + 1) || (nullptr == dest))
+  {
+    *olen = n + 1;
+    return -1;
+  }
+
+  n = (slen / 3) * 3;
+
+  for (i = 0, p = dest; i < n; i += 3)
+  {
+    C1 = *src++;
+    C2 = *src++;
+    C3 = *src++;
+
+    *p++ = base64_enc_map[(C1 >> 2) & 0x3F];
+    *p++ = base64_enc_map[(((C1 & 3) << 4) + (C2 >> 4)) & 0x3F];
+    *p++ = base64_enc_map[(((C2 & 15) << 2) + (C3 >> 6)) & 0x3F];
+    *p++ = base64_enc_map[C3 & 0x3F];
+  }
+
+  if (i < slen)
+  {
+    C1 = *src++;
+    C2 = ((i + 1) < slen) ? *src++ : 0;
+
+    *p++ = base64_enc_map[(C1 >> 2) & 0x3F];
+    *p++ = base64_enc_map[(((C1 & 3) << 4) + (C2 >> 4)) & 0x3F];
+
+    if ((i + 1) < slen)
+    {
+      *p++ = base64_enc_map[((C2 & 15) << 2) & 0x3F];
+    }
+    else if (padding_char)
+    {
+      *p++ = padding_char;
+    }
+
+    if (padding_char)
+    {
+      *p++ = padding_char;
+    }
+  }
+
+  *olen = static_cast<size_t>(p - dest);
+  *p    = 0;
+
+  return (0);
+}
+
+static inline int Base64EscapeInternal(std::string &dest,
+                                       const unsigned char *src,
+                                       size_t slen,
+                                       Base64EscapeChars &base64_enc_map,
+                                       unsigned char padding_char)
+{
+  size_t olen = 0;
+  Base64EscapeInternal(nullptr, 0, &olen, src, slen, base64_enc_map, padding_char);
+  dest.resize(olen);
+
+  if (nullptr == src || 0 == slen)
+  {
+    return 0;
+  }
+
+  int ret = Base64EscapeInternal(reinterpret_cast<unsigned char *>(&dest[0]), dest.size(), &olen,
+                                 src, slen, base64_enc_map, padding_char);
+#if defined(HAVE_GSL)
+  Expects(0 != ret || dest.size() == olen + 1);
+#else
+  assert(0 != ret || dest.size() == olen + 1);
+#endif
+  // pop back last zero
+  if (!dest.empty() && *dest.rbegin() == 0)
+  {
+    dest.resize(dest.size() - 1);
+  }
+  return ret;
+}
+
+// Base64Escape()
+//
+// Encodes a `src` string into a base64-encoded 'dest' string with padding
+// characters. This function conforms with RFC 4648 section 4 (base64) and RFC
+// 2045. See also CalculateBase64EscapedLen().
+static void Base64Escape(nostd::string_view src, std::string *dest)
+{
+  if (nullptr == dest)
+  {
+    return;
+  }
+
+  Base64EscapeInternal(*dest, reinterpret_cast<const unsigned char *>(src.data()), src.size(),
+                       kBase64CharsBasic, '=');
+}
+
```

Comment:
Just curious, why to implement this and not use from abseil/google. Maintaining this code would be issue in future ?